stackit/internal/services/secretsmanager/user/resource.go

```diff
@@ -0,0 +1,410 @@
+	resp.TypeName = req.ProviderTypeName + "_secretsmanager_user"
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *userResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerData, ok := req.ProviderData.(core.ProviderData)
+	if !ok {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Expected configure type stackit.ProviderData, got %T", req.ProviderData))
+		return
+	}
+
+	var apiClient *secretsmanager.APIClient
+	var err error
+	if providerData.SecretsManagerCustomEndpoint != "" {
+		apiClient, err = secretsmanager.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.SecretsManagerCustomEndpoint),
+		)
+	} else {
+		apiClient, err = secretsmanager.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the resource configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "Secrets Manager user client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *userResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":          "Secrets Manager user resource schema. Must have a `region` specified in the provider configuration.",
+		"id":            "Terraform's internal resource identifier. It is structured as \"`project_id`,`instance_id`,`user_id`\".",
+		"user_id":       "The user's ID.",
+		"instance_id":   "ID of the Secrets Manager instance.",
+		"project_id":    "STACKIT Project ID to which the instance is associated.",
+		"description":   "A user chosen description to differentiate between multiple users. Can't be changed after creation.",
+		"write_enabled": "If true, the user has writeaccess to the secrets engine.",
+		"username":      "An auto-generated user name.",
+		"password":      "An auto-generated password.",
+	}
+
+	resp.Schema = schema.Schema{
+		Description: descriptions["main"],
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: descriptions["id"],
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"user_id": schema.StringAttribute{
+				Description: descriptions["user_id"],
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
```

Comment:
Do we want/need validators for a computed only field?

Reply:
Yes, this is telling TF that this value won't change on updated, so TF can show the current value on `terraform plan` instead of "(known after apply)"